@aspnet/signalr has a hard dependency on node packages even though it is designed for usage in browser.

### Describe the bug
`@aspnet/signalr` TypeScript definition files reference node types.

### To Reproduce
Steps to reproduce the behavior:
1. `npm init`
1. `npm install @aspnet/signalr`
1. Add `tsconfig.json`
	```typescript
	{
		"compilerOptions": {
			"target": "es2015",
			"module": "es2015",
			"moduleResolution": "node",
			"outDir": "output",
			"rootDir": "source",
			"strict": true,
			"noEmitOnError": true,
			"lib": [ "dom", "es2015" ]
		},
		"include": [
			"source/**/*.ts"
		]
	}
	```
1. Create a file `source/index.ts`
    import * as signalR from '@aspnet/signalr'
    signalR
1. `npx tsc`
1. Notice the two errors:
	```
	node_modules/@aspnet/signalr/dist/esm/IHubProtocol.d.ts:1:23 - error TS2688: Cannot find type definition file for 'node'.
	1 /// <reference types="node" />
	                        ~~~~
	node_modules/@aspnet/signalr/dist/esm/IHubProtocol.d.ts:130:49 - error TS2580: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i @types/node`.
	130     parseMessages(input: string | ArrayBuffer | Buffer, logger: ILogger): HubMessage[];
	                                                    ~~~~~~
	Found 4 errors.
	```

### Expected behavior
Expected `@aspnet/signalr` to be usable in a browser without having to polyfill node.

--- a/src/SignalR/clients/ts/signalr-protocol-msgpack/package.json
+++ b/src/SignalR/clients/ts/signalr-protocol-msgpack/package.json
@@ -47,7 +47,6 @@
     "msgpack5": "^4.0.2"
   },
   "devDependencies": {
-    "@types/msgpack5": "^3.4.1",
-    "buffer": "^5.0.8"
+    "@types/msgpack5": "^3.4.1"
   }
 }

--- a/src/SignalR/clients/ts/signalr-protocol-msgpack/src/MessagePackHubProtocol.ts
+++ b/src/SignalR/clients/ts/signalr-protocol-msgpack/src/MessagePackHubProtocol.ts
@@ -1,7 +1,6 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-import { Buffer } from "buffer";
 import * as msgpack5 from "msgpack5";
 
 import { MessagePackOptions } from "./MessagePackOptions";
@@ -51,13 +50,13 @@ export class MessagePackHubProtocol implements IHubProtocol {
 
     /** Creates an array of HubMessage objects from the specified serialized representation.
      *
-     * @param {ArrayBuffer | Buffer} input An ArrayBuffer or Buffer containing the serialized representation.
+     * @param {ArrayBuffer} input An ArrayBuffer containing the serialized representation.
      * @param {ILogger} logger A logger that will be used to log messages that occur during parsing.
      */
-    public parseMessages(input: ArrayBuffer | Buffer, logger: ILogger): HubMessage[] {
+    public parseMessages(input: ArrayBuffer, logger: ILogger): HubMessage[] {
         // The interface does allow "string" to be passed in, but this implementation does not. So let's throw a useful error.
-        if (!(input instanceof Buffer) && !(isArrayBuffer(input))) {
-            throw new Error("Invalid input for MessagePack hub protocol. Expected an ArrayBuffer or Buffer.");
+        if (!(isArrayBuffer(input))) {
+            throw new Error("Invalid input for MessagePack hub protocol. Expected an ArrayBuffer.");
         }
 
         if (logger === null) {
@@ -108,7 +107,8 @@ export class MessagePackHubProtocol implements IHubProtocol {
         }
 
         const msgpack = msgpack5(this.messagePackOptions);
-        const properties = msgpack.decode(Buffer.from(input));
+        // To avoid using the Buffer type we cast to 'any'. msgpack5 works with Uint8Array's
+        const properties = msgpack.decode(input as any);
         if (properties.length === 0 || !(properties instanceof Array)) {
             throw new Error("Invalid payload.");
         }

--- a/src/SignalR/clients/ts/signalr-protocol-msgpack/webpack.config.js
+++ b/src/SignalR/clients/ts/signalr-protocol-msgpack/webpack.config.js
@@ -8,8 +8,5 @@ module.exports = baseConfig(__dirname, "signalr-protocol-msgpack", {
     externals: {
         msgpack5: "msgpack5",
         "@microsoft/signalr": "signalR"
-    },
-    alias: {
-        buffer: path.resolve(__dirname, "node_modules", "buffer", "index.js"),
-    },
+    }
 });

--- a/src/SignalR/clients/ts/signalr/src/HandshakeProtocol.ts
+++ b/src/SignalR/clients/ts/signalr/src/HandshakeProtocol.ts
@@ -28,7 +28,7 @@ export class HandshakeProtocol {
         let messageData: string;
         let remainingData: any;
 
-        if (isArrayBuffer(data) || (typeof Buffer !== "undefined" && data instanceof Buffer)) {
+        if (isArrayBuffer(data)) {
             // Format is binary but still need to read JSON text from handshake response
             const binaryData = new Uint8Array(data);
             const separatorIndex = binaryData.indexOf(TextMessageFormat.RecordSeparatorCode);

--- a/src/SignalR/clients/ts/signalr/src/IHubProtocol.ts
+++ b/src/SignalR/clients/ts/signalr/src/IHubProtocol.ts
@@ -157,10 +157,10 @@ export interface IHubProtocol {
      *
      * If {@link @microsoft/signalr.IHubProtocol.transferFormat} is 'Text', the `input` parameter must be a string, otherwise it must be an ArrayBuffer.
      *
-     * @param {string | ArrayBuffer | Buffer} input A string, ArrayBuffer, or Buffer containing the serialized representation.
+     * @param {string | ArrayBuffer} input A string or ArrayBuffer containing the serialized representation.
      * @param {ILogger} logger A logger that will be used to log messages that occur during parsing.
      */
-    parseMessages(input: string | ArrayBuffer | Buffer, logger: ILogger): HubMessage[];
+    parseMessages(input: string | ArrayBuffer, logger: ILogger): HubMessage[];
 
     /** Writes the specified {@link @microsoft/signalr.HubMessage} to a string or ArrayBuffer and returns it.
      *

--- a/src/SignalR/clients/ts/signalr/webpack.config.js
+++ b/src/SignalR/clients/ts/signalr/webpack.config.js
@@ -7,7 +7,7 @@ module.exports = env => baseConfig(__dirname, "signalr", {
     target: env && env.platform ?  env.platform : undefined,
     platformDist: env && env.platform ?  env.platform : undefined,
     externals: [
-        "websocket",
+        "ws",
         "eventsource",
         "node-fetch",
         "abort-controller",

